### PR TITLE
MM/XPMEM: Multiple fixes.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by yosefe** [comment]
Fixes errors in #760

**Comment by shamisp** [comment]
Did you add extra testing that exposed the alignment bug ?

File: src/uct/sm/mm/mm_pd.c [hunk 1]
```diff
@@ -72,7 +72,7 @@ ucs_status_t uct_mm_mem_reg(uct_pd_h pd, void *address, size_t length,
     ucs_status_t status;
     uct_mm_seg_t *seg;
 
-    seg = ucs_malloc(sizeof(*seg), "mm_seg");
+    seg = ucs_calloc(1, sizeof(*seg), "mm_seg");
```
Comment (shamisp) [inline src/uct/sm/mm/mm_pd.c]: why 1 ?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1254/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1866/ for details.

File: src/uct/sm/mm/mm_pd.c (hunk 1 above)
Comment (yosefe) [inline src/uct/sm/mm/mm_pd.c]: because nmemb parameter to calloc should be 1 in this case

**Comment by yosefe** [comment]
@shamisp After adding UCP multi-transport code, UCP started using xpmem, and this exposed the bug (on Matt's setup). I've installed xpmem on a virtual machine and reproduced that as well. This PR fixes all those bugs.

**Comment by shamisp** [comment]
@yosefe - probably we don't have any testbed with xpmem. Something to fix. I will go over the changes later on.

My understating that the "offset"  and "initialization" is the bug and all the rest is styling ?

**Comment by yosefe** [comment]
also error check, and valgrind false-positives. see commit message for details.

**Comment by MattBBaker** [comment]
I tested this out on Titan, this combined with UCP multi-transport code works 100% on Titan. :+1:

**Comment by shamisp** [comment]
I still have to test it on ARM

File: src/uct/sm/mm/mm_pd.c (hunk 1 above)
Comment (shamisp) [inline src/uct/sm/mm/mm_pd.c]: oh, was looking at it and thinking about memset. Ignore for now.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1258/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1262/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
👍

**Pull request merged**